### PR TITLE
Add anchor kwarg to blit()

### PR DESCRIPTION
**Pull request by novialriptide**
Fixes: https://github.com/pygame/pygame/issues/3202

@Starbuck5 and I had a conversation on how to optimize this better, but we couldn't get it to be faster than the Python code using `get_rect()`. If someone has any idea on how to optimize this, that'll be great.

```
pygame 2.1.3.dev5 (SDL 2.0.22, Python 3.9.0)
Hello from the pygame community. https://www.pygame.org/contribute.html
## ANCHOR KWARG ##
no kwargs: 0.2553597000000001 seconds
topleft: 0.4961052 seconds
topright: 0.6383643999999999 seconds
bottomleft: 0.6494785000000001 seconds
bottomright: 0.6352516000000001 seconds
midleft: 0.6348135999999998 seconds
midright: 0.6555408000000003 seconds
midtop: 0.6400424000000005 seconds
midbottom: 0.6353241000000001 seconds
center: 0.6293594000000002 seconds
PS C:\Users\novia\Desktop\blit kwargs test> python test3.py
pygame 2.1.3.dev5 (SDL 2.0.22, Python 3.9.0)
Hello from the pygame community. https://www.pygame.org/contribute.html
## GET_RECT() ##
no rect: 0.2463491 seconds
topleft: 0.40525160000000005 seconds
topright: 0.4099286999999999 seconds
bottomleft: 0.40654049999999997 seconds
bottomright: 0.4106080000000001 seconds
midleft: 0.41274900000000025 seconds
midright: 0.4012313999999999 seconds
midtop: 0.4056690999999999 seconds
midbottom: 0.40756900000000007 seconds
center: 0.4032928 seconds
```

### CODE REVIEW

File: src_c/surface.c [hunk 1]
```diff
@@ -1863,6 +1865,52 @@ surf_blit(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
...
+    make this work. Using `PyObject_SetAttr()` on a
+    pgRect in C is actually slower than this method.
+
+    This code is also slower than Python code itself.
+
+    TODO: Optimize or cleanup implementation.
+    */
+    if (anchor_text == NULL) {
+    }
+    else if (strcmp(anchor_text, "topleft") == 0) {
```
Comment (itzpr3d4t0r) [inline src_c/surface.c:1880]: is it worth some performance if we shorten the anchor names to just to characters? like `topleft = tl`, `bottomright = br`?

Comment (novialriptide) [reply src_c/surface.c:1880]: Thinking more about this, I'm not so sure about this. Any thoughts? @Starbuck5 @ankith26 @illume

Comment (nthykier) [inline src_c/surface.c:1880]: A classic C stunt here might be to abuse `switch`.  Something like:

```c
if (anchor_text != NULL) {
  int valid = 0;
  switch(anchor_text[0]) {
    case 't':
      if (strcmp(anchor_text, "topleft") == 0) {
         ...:
         valid = 1;
      } else if (strcmp(anchor_text, "topright") == 0) {
         ...:
         valid = 1;
      }
      break;
    case 'b':
      ...;
      break;
    case 'm':
      ...;
      break;
    case 'c':
      ...;
      break;
    ....
  }
  if (!valid) {
    return RAISE(PyExc_KeyError, "invalid anchor");
  }
}
```

But then, one might hope that the compiler was smart enough to do this already.

**Comment by novialriptide** [comment]
Could someone add a don't merge tag to this PR?

**Comment by novialriptide** [comment]
New idea: `pygame.Surface.pblit()`

**Pull request closed without merging**